src/solver/expressions/include/antares/solver/expressions/nodes/BinaryNode.h

```diff
@@ -0,0 +1,32 @@
+/*
+** Copyright 2007-2024, RTE (https://www.rte-france.com)
+** See AUTHORS.txt
+** SPDX-License-Identifier: MPL-2.0
+** This file is part of Antares-Simulator,
+** Adequacy and Performance assessment for interconnected energy networks.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the Mozilla Public Licence 2.0 as published by
+** the Mozilla Foundation, either version 2 of the License, or
+** (at your option) any later version.
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** Mozilla Public Licence 2.0 for more details.
+**
+** You should have received a copy of the Mozilla Public Licence 2.0
+** along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+*/
+#pragma once
+
+#include <antares/solver/expressions/nodes/ParentNode.h>
+
+namespace Antares::Solver::Nodes
+{
+class BinaryNode: public ParentNode<2>
```

Comment:
I don't agree with you, the intention behind BinaryNode and UnaryNode is clear, to represent nodes containing 1 or 2 nodes. The client of these classes knows exactly how to build them, and how to use them with the[] operator.

Reply:
_"n = 1 : using a [] operator is overkill. So any Node class that contains another Node should not inherit from ParentNode<1>"_ why would someone inherit from ParentNode<1> if he/she seeks to store more than one node?

Reply:
I disagree as well. This allows to eliminate a lot of common code and avoid duplication.

Reply:
I meant : 
- if a class aims at containing just one other node (like **NegationNode**), the benefit for this class to inherit from **ParentNode<1>** is very small. Indeed, in general, the only benefit for a class to inherit from **ParentNode** is to reuse the **[] operator**. In this particular case, it's not a benefit : we always use the first element (index 0).
- So we may think that the benefit of reusing the **[] operator** of **ParentNode** could be higher if n > 1, right ? 
- But, as said above, n = 3 does not require the reuse of this operator either.
- Then if the class **ParentNode** is only useful for case n = 2, its interest is limited, and this parent class is probably not useful. If you add to this the fact that it makes class hierarchy grow, it can feel like removing it.